src/extension/wmts-multi-dimensional/src/test/java/org/geoserver/gwc/wmts/MultiDimensionalExtensionTest.java

```diff
@@ -1264,4 +1270,21 @@ public void testRasterCustomGetDomainValues() throws Exception {
         assertXpathEvaluatesTo(CUSTOM_DIMENSION_NAME, "/md:DomainValues/ows:Identifier", dom);
         assertXpathEvaluatesTo("CustomDimValueA", "/md:DomainValues/md:Domain", dom);
     }
+
+    @Test
+    public void testTileLayerNotInstanceOfGeoServerTileLayer() {
+        TileLayer tileLayer = mock(TileLayer.class);
+        TileLayerDispatcher tld = mock(TileLayerDispatcher.class);
+        XMLBuilder xml = new XMLBuilder(new StringBuilder());
+        MultiDimensionalExtension extension =
+                new MultiDimensionalExtension(getGeoServer(), getWMS(), getCatalog(), tld);
+        // No exception is thrown when the layer isn't a GeoServerTileLayer
+        boolean layerEncoded = true;
+        try {
+            extension.encodeLayer(xml, tileLayer);
```

Comment:
No need for all the checks, a test can throw an exception, it's a legit way to fail. Just add a note stating that the line used to throw an exception.